airbyte-integrations/connectors/source-monday/components.py

```diff
@@ -1,23 +1,132 @@
+                        continue
+
+                    if record.get("entity") == "pulse" and json_data.get("pulse_id"):
+                        new_record.update({"pulse_id": json_data.get("pulse_id")})
+
+                    if record.get("entity") == "board" and json_data.get("board_id"):
+                        new_record.update({"board_id": json_data.get("board_id")})
+
+                    yield new_record
+
+
+@dataclass
+class MondayIncrementalItemsExtractor(RecordExtractor):
+    """
+    Record extractor that searches a decoded response over a path defined as an array of fields.
+    """
+
+    field_path: List[Union[InterpolatedString, str]]
+    config: Config
+    parameters: InitVar[Mapping[str, Any]]
+    field_path_pagination: List[Union[InterpolatedString, str]] = field(default_factory=list)
+    decoder: Decoder = field(default_factory=lambda: JsonDecoder(parameters={}))
+
+    def __post_init__(self, parameters: Mapping[str, Any]):
+        # Convert string paths to InterpolatedString for both field_path and field_path_pagination
+        self._field_path = [InterpolatedString.create(p, parameters=parameters) if isinstance(p, str) else p for p in self.field_path]
+        self._field_path_pagination = [
+            InterpolatedString.create(p, parameters=parameters) if isinstance(p, str) else p for p in self.field_path_pagination
+        ]
+
+    def _try_extract_records(
+        self, response: requests.Response, field_path: List[Union[InterpolatedString, str]]
+    ) -> Iterable[Mapping[str, Any]]:
+        for body in self.decoder.decode(response):
+            if len(field_path) == 0:
+                extracted = body
+            else:
+                path = [p.eval(self.config) for p in field_path]
+                if "*" in path:
+                    extracted = dpath.values(body, path)
+                else:
+                    extracted = dpath.get(body, path, default=[])
+
+            if extracted:
+                if isinstance(extracted, list) and None in extracted:
+                    logger.warning(f"Record with null value received; errors: {body.get('errors')}")
+                    yield from (x for x in extracted if x)
+                else:
+                    yield from extracted if isinstance(extracted, list) else [extracted]
+
+    def extract_records(self, response: requests.Response) -> Iterable[Mapping[str, Any]]:
+        # Try primary field path
+        has_records = False
+        for record in self._try_extract_records(response, self._field_path):
+            has_records = True
+            yield record
+
+        # Fallback to pagination path if no records and path exists
+        if not has_records and self._field_path_pagination:
+            yield from self._try_extract_records(response, self._field_path_pagination)
 
 
 @dataclass
 class MondayGraphqlRequester(HttpRequester):
     NEXT_PAGE_TOKEN_FIELD_NAME = "next_page_token"
 
+    schema_loader: InlineSchemaLoader = None
```

Comment:
Nit: because this is a required field, we don't need to have `= None`. It's actually probably a type mismatch if we ran mypy since The type is not `Optional[InlineSchemaLoader`. So this should just be `schema_loader: InlineSchemaLoader`